base/poco/NetSSL_OpenSSL/src/SecureSocketImpl.cpp

```diff
@@ -424,15 +423,25 @@ long SecureSocketImpl::verifyPeerCertificateImpl(const std::string& hostName)
 	Context::VerificationMode mode = _pContext->verificationMode();
 	if (mode == Context::VERIFY_NONE || !_pContext->extendedCertificateVerificationEnabled() ||
 	    (mode != Context::VERIFY_STRICT && isLocalHost(hostName)))
-	{
 		return X509_V_OK;
-	}
 
-	X509* pCert = SSL_get_peer_certificate(_pSSL);
+	X509* pCert = SSL_get1_peer_certificate(_pSSL);
 	if (pCert)
 	{
-		X509Certificate cert(pCert);
-		return cert.verify(hostName) ? X509_V_OK : X509_V_ERR_APPLICATION_VERIFICATION;
+        if (X509_check_host(pCert, hostName.c_str(), hostName.length(), 0, nullptr) == 1)
```

Comment:
Wondering, was the code below copy pasted from elsewhere?

Reply:
https://github.com/ClickHouse/ClickHouse/blob/master/base/poco/NetSSL_OpenSSL/src/X509Certificate.cpp#L136-L148